backends/c-hlsl_xbox/evaluator/AntaresHlslLib/D3D12APIWrapper.cpp

```diff
@@ -379,6 +379,13 @@ void* dxModuleLoad(const char* module_src)
             return nullptr;
         std::string _((std::istreambuf_iterator<char>(t)), std::istreambuf_iterator<char>());
         source = std::move(_);
+        if (source.find('\r') != std::string::npos)
+        {
+            std::string res = source;
+            std::string::iterator end_pos = std::remove(res.begin(), res.end(), '\r');
+            res.erase(end_pos, res.end());
+            source = std::move(res);
+        }
         module_src = source.c_str();
     }
 

```

Comment:
else branch from most-outer branch also need this checking.
To avoid redundant logic, it would be better to put this logic after this branch:
```cpp
  if (strncmp(module_src, proto, sizeof(proto) - 1) == 0) {
    ..
  }
+ source = remove_chars(module_src, '\n');
  module_src = source.c_str();
```